code/src/pages/cart/components/CartItem.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from "react";
+
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+import DeleteIcon from "@mui/icons-material/Delete";
+import RemoveCircleOutlineIcon from "@mui/icons-material/RemoveCircleOutline";
+
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+
+import formatPrice from "@/utils/format-price/formatPrice";
+
+import "@/pages/cart/components/CartItem.scss";
+
+type CartItemProps = {
+  item: {
+    image: string;
+    name: string;
+    productPrice: number;
+    quantity: number;
+    calculatedPrice: number;
+  };
+  onQuantityChange: (quantity: number) => void;
+  onDelete: () => void;
+};
+
+const CartItem = ({ item, onQuantityChange, onDelete }: CartItemProps) => {
+  const [quantity, setQuantity] = useState<string>(item.quantity.toString());
+
+  useEffect(() => {
+    setQuantity(item.quantity.toString());
+  }, [item.quantity]);
+
+  const handleDecrement = () => {
+    if (item.quantity > 1) {
+      onQuantityChange(item.quantity - 1);
+    }
+  };
+
+  const handleIncrement = () => {
+    onQuantityChange(item.quantity + 1);
+  };
+
+  const handleInputBlur = () => {
+    if (quantity === "") {
+      setQuantity(item.quantity.toString());
+    } else {
+      onQuantityChange(parseInt(quantity, 10));
+    }
+  };
+
+  return (
+    <AppBox className="spa-cart-item">
+      <img src={item.image} alt={item.name} className="spa-cart-item__image" />
+      <AppBox className="spa-cart-item__details">
+        <AppTypography className="spa-cart-item__title" variant="h3">
+          {item.name}
+        </AppTypography>
+        <AppTypography component="p">
+          {formatPrice(item.productPrice)}
+        </AppTypography>
+      </AppBox>
+      <AppBox className="spa-cart-item__quantity-selector">
+        <AppBox
+          onClick={handleDecrement}
+          className="spa-cart-item__quantity-block"
+        >
+          <RemoveCircleOutlineIcon />
+        </AppBox>
+        <input
+          type="text"
+          value={quantity}
+          onBlur={handleInputBlur}
+          className="spa-cart-item__quantity-input"
+        />

```

Comment:
You may have error in console because you don't use `onChange` handler for input

Reply:
> You may have error in console because you don't use `onChange` handler for input

Input logic with changing quantity will be added further I wrote it in the description to pr